### PR TITLE
Replacing Travis with CircleCI

### PR DESCRIPTION
**Pull request by moekify**
There are some quirks to note.

### It uses a Java base image
The gem that we use to manage the s3_website deployment does not support more recent versions of Java and it is partially written in Scala.

### There are a LOT of ENV vars to setup
This is just pretty much minimally translating what is there for Travis, it does seem like there are a number of env vars which are unused which would be nice to cleanup, or use the .env.* stuff that Gatsby supports.
This is mostly from what I can tell an artifact of copy-pasting the deployment from our old static-pages into the 'new' website and just going with it. We will hopefully review our FE repo deployments more wholistically soon so I've left this as-is for now.

Of particular note is the nonsense this commit (https://github.com/honeypotio/website/pull/131/commits/4558027f5fa1104b0fc60eaa7fa1e5475f1193f4) has to deal with because of one of our dependent gems.

Related ticket: https://app.clickup.com/t/a4xpx6

### CODE REVIEW

**Comment by jneurock** [comment]
Small thing: I think this PR has the wrong title.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      - run:
+          name: Setup nvm
+          command: |
+              export NVM_DIR="$HOME/.nvm"
+              [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+              echo 'export NVM_DIR=$HOME/.nvm' >> $BASH_ENV
+              echo 'source $NVM_DIR/nvm.sh' >> $BASH_ENV
+      - run:
+          name: install node
+          command: nvm install
```
Comment (jneurock) [inline .circleci/config.yml:89]: Not a blocker: Are these system level dependencies cached or are these steps relatively quick? If not, can we find a Docker image that includes Java 8, Ruby 2.5.3, etc.? Or does it not matter for now, if we're thinking we want to revisit how we deploy this soon and we don't open PRs on this repo much?

**Review by jneurock — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (moekify) [reply .circleci/config.yml:89]: Overall duration of the pipeline is currently a bit above 3 minutes. The ruby install takes like 24 seconds. 
I think we can ignore this for now until we revisit this altogether as @Ryman suggested

https://app.circleci.com/pipelines/github/honeypotio/website/35/workflows/e3a526b3-8039-4ae6-bfa0-fe6a3892f187/jobs/37

Comment (jneurock) [reply .circleci/config.yml:89]: 👍

**Comment by moekify** [comment]
Task linked: [CU-a4xpx6 Migrate from Travis CI](https://app.clickup.com/t/a4xpx6)

**Pull request merged**